src/metabase/models/params.clj

```diff
@@ -100,25 +100,31 @@
        field-ids->param-field-values-ignoring-current-user
        params.field-values/field-id->field-values-for-current-user) param-field-ids)))
 
-(defn- template-tag->field-form
+(defn- find-template-tag
   "Fetch the `:field` clause from `dashcard` referenced by `template-tag`.
 
-    (template-tag->field-form [:template-tag :company] some-dashcard) ; -> [:field 100 nil]"
+    (find-template-tag [:template-tag :company] some-dashcard) ; -> [:field 100 nil]"
   [[_ tag] card]
-  (get-in card [:dataset_query :native :template-tags (u/qualified-name tag) :dimension]))
+  (get-in card [:dataset_query :native :template-tags (u/qualified-name tag)]))
 
-(mu/defn param-target->field-clause :- [:maybe mbql.s/Field]
+(mu/defn param-target->template-tag :- [:maybe mbql.s/TemplateTag]
```

Comment:
I don't think we should change this function to make it return a template tag.
As I said, there are multiple types of template tags, and a template tag with `:dimension` is one of them, and in product terms, it's referred to "field filter"

it makes the usage of this function confusing. takes [this](https://github.com/metabase/metabase/blob/34db49580ff698a4cf5385bf2ffebade44a1fb99/src/metabase/api/card.clj#L1099) for example, under that context it's not clear why we call `:dimension` on a template when we don't know for sure it's a FieldFilter template tag.

Reply:
Hmmm... right, but if I just return a field clause, then there is no way to know any options defined on it. Or are you suggesting to return "something", like `{:field ... :options ...}`? I was trying to keep it closer to an already defined entities.

Reply:
how about we not use it at all? I think we can re-think the current `api.dashboard/mappings->field-ids` function, 

I don't think it needs to call this function, I think what it's trying to do is :
- if the card is native => get the template tag field id that the template tag map to
- if the card is mbql => get the id from `result_metadata`

maybe we can simplified this for our own needs.